catalog/security/catalog-security-plugin/src/main/java/ddf/catalog/security/plugin/SecurityPlugin.java

```diff
@@ -49,17 +49,28 @@ public SecurityPlugin(SubjectIdentity subjectIdentity) {
     this.subjectIdentity = subjectIdentity;
   }
 
+  /**
+   * Sets the {@link Metacard#POINT_OF_CONTACT} attribute on a {@link Metacard} if it does not
+   * contain one already and the metacard is a resource metacard. From a legacy perspective, a
+   * metacard with no tags is considered a resource metacard.
+   *
+   * @param input the {@link CreateRequest} containing {@link Metacard}s to process
+   * @return the original {@link CreateRequest}
+   */
   @Override
-  public CreateRequest processPreCreate(CreateRequest input) throws StopProcessingException {
+  public CreateRequest processPreCreate(CreateRequest input) {
     ddf.security.Subject subject = setSubjectOnRequestProperties(input);
 
     if (input.getMetacards() != null && subject != null) {
       input
           .getMetacards()
+          .stream()
+          .filter(metacard -> metacard.getAttribute(Metacard.POINT_OF_CONTACT) == null)
           .forEach(
               metacard -> {
                 String id = subjectIdentity.getUniqueIdentifier(subject);
```

Comment:
✏️ I would really prefer to see this lambda extracted to a method to keep the stream processing logic and the lower-level "do this to each metacard" logic separated.

Could also consider extracting some of the filtering logic - the check for empty tags or tags that include the `DEFAULT_TAG` - above the `forEach`

Something like this (this might not be quite right):
```java
input
    .getMetacards()
    .stream()
    .filter(metacard -> metacard.getAttribute(Metacard.POINT_OF_CONTACT) == null)
    .filter(metacard -> metacard.getTags().isEmpty || metacard.getTags().contains(Metacard.DEFAULT_TAG))
    .forEach(metacard -> doit(metacard, subject));
```
with
```java
  private void doit(Metacard metacard, ddf.security.Subject subject) {
    String id = subjectIdentity.getUniqueIdentifier(subject);
    if (StringUtils.isNotBlank(id)) {
      metacard.setAttribute(new AttributeImpl(Metacard.POINT_OF_CONTACT, id));
    }
  }
```